### PR TITLE
Create a generic method for mapping API endpoints

### PR DESCRIPTION
**Pull request by mstypsan**
### Summary & Motivation

This pull request will Map all of the application's endpoints by scanning all of the classes that implement the `IEndpoints` interface.

### Checklist

- [x] I have added a Label to the pull-request
- [x] I have added tests, and done manual regression tests
- [x] I have updated the documentation, if necessary

### CODE REVIEW

File: application/shared-kernel/ApiCore/ApiCoreConfiguration.cs [hunk 1]
```diff
@@ -164,6 +179,14 @@ public static WebApplication AddApiCoreConfiguration<TDbContext>(this WebApplica
 
         // Add test-specific endpoints when running tests, such as /api/throwException
         app.MapTestEndpoints();
```
Comment (tjementum) [inline application/shared-kernel/ApiCore/ApiCoreConfiguration.cs:181]: I guess the `TestEndpoints`, `HealtEndpoints`, and `TrackEndpoints` in the ApiCore, could/should also be implemented using the `IEndpoints`. We

The `HealtEndpoints` are in the `ServiceDefaultsExtensions` class. We should extract these to a separate class and move them to the `/Endpoints` folder.

File: application/account-management/Api/Program.cs [hunk 2]
```diff
@@ -23,10 +21,6 @@
 // Add common configuration for all APIs like Swagger, HSTS, DeveloperExceptionPage, and run EF database migrations.
 app.AddApiCoreConfiguration<AccountManagementDbContext>();
 
-app.MapAccountRegistrationsEndpoints();
-app.MapTenantEndpoints();
-app.MapUserEndpoints();
```
Comment (tjementum) [inline application/account-management/Api/Program.cs:28]: This is awesome :)

**Review by tjementum — CHANGES_REQUESTED** [review]
Love it :) I made a few suggestions to improvements.

File: application/shared-kernel/ApiCore/ApiCoreConfiguration.cs (hunk 1 above)
Comment (mstypsan) [reply application/shared-kernel/ApiCore/ApiCoreConfiguration.cs]: Done

**Comment by mstypsan** [comment]
> Love it :) I made a few suggestions to improvements.

@tjementum give it a look again

**Review by tjementum — APPROVED** [review]

**Pull request merged**